File: src/cache.ts
import LRU from "lru-cache";

type Key = string;
type Val<T> = { value: T; ts: number; stale?: boolean };

export function createCache<T>(ttlMs: number) {
  const lru = new LRU<Key, Val<T>>({ max: 64, ttl: ttlMs });
  return {
    get: (k: Key) => lru.get(k)?.value,
    set: (k: Key, v: T) => lru.set(k, { value: v, ts: Date.now() }),
    has: (k: Key) => lru.has(k)
  };
}

File: src/cache.py
"""Simple cache implementation using cachetools TTLCache."""
import time
from typing import TypeVar, Generic, Optional
from cachetools import TTLCache


T = TypeVar("T")


class Cache(Generic[T]):
    """Generic TTL-based cache wrapper."""
    
    def __init__(self, ttl_ms: int):
        """Initialize cache with TTL in milliseconds."""
        self._cache: TTLCache = TTLCache(maxsize=64, ttl=ttl_ms / 1000.0)
    
    def get(self, key: str) -> Optional[T]:
        """Get value from cache."""
        return self._cache.get(key)
    
    def set(self, key: str, value: T) -> None:
        """Set value in cache."""
        self._cache[key] = value
    
    def has(self, key: str) -> bool:
        """Check if key exists in cache."""
        return key in self._cache


def create_cache(ttl_ms: int) -> Cache:
    """Factory function to create a cache instance."""
    return Cache(ttl_ms)
